Tuesdays/Week-Class/Inheratance/Assets/Scripts/TrapController.cs

```diff
@@ -3,18 +3,43 @@
 using UnityEngine;
 
 public class TrapController : MonoBehaviour{
-	[SerializeField] private Trap[] trapList;
     [SerializeField] private Transform[] spawnPoints;
-
+	[SerializeField] private Trap _trapPrefab;
+	private List<Trap.TrapDelegate> _trapBehaviours;
+	[SerializeField] TrapSprite[] _trapSprites;
+	[SerializeField] LayerMask[] _damageTarget;
+	
 	void Start(){
-		int rand;
-
-		foreach (Transform t in spawnPoints) {
-			rand = Random.Range(0, trapList.Length);
-			Trap randTrap = trapList[rand];
-			Instantiate(randTrap,t.position, t.rotation);
+		_trapBehaviours = new List<Trap.TrapDelegate>();
+		_trapBehaviours.Add(TrapBehaviours.DoDamage);
```

Comment:
would of preferred if trap behaviors had the registration logic.
by exposing `GetallTrapBehaviours` function or something